index.js

```diff
@@ -1,293 +1,324 @@
-tickers['BitStampUSD'] = null;
-tickers['BTCeUSD'] = null;
-tickers['KrakenUSD'] = null;
-tickers['CoinDeskUSD'] = null;
-tickers['CoinbaseUSD'] = null;
-tickers['CampBXUSD'] = null;
-tickers['BitPayUSD'] = null;
-tickers['TheRockTradingUSD'] = null;
-tickers['BitFinexUSD'] = null;
-tickers['BTCeEUR'] = null;
-tickers['KrakenEUR'] = null;
-tickers['CoinDeskEUR'] = null;
-tickers['BitPayEUR'] = null;
-tickers['BitonicEUR'] = null;
-tickers['Bitcoin-CentralEUR'] = null;
-tickers['TheRockTradingEUR'] = null;
-tickers['CoinDeskGBP'] = null;
-tickers['LocalbitcoinsGBP'] = null;
-tickers['BittyliciousGBP'] = null;
-tickers['BitPayGBP'] = null;
-tickers['BitcurexPLN'] = null;
-tickers['CaVirTexCAD'] = null;
-tickers['BTCChinaCNY'] = null;
-tickers['MercadoBitcoinBRL'] = null;
-tickers['BTCTurkTRY'] = null;
-tickers['BitcoinVenezuelaVEF'] = null;
-tickers['LocalbitcoinsVEF'] = null;
-tickers['CoinbaseVEF'] = null;
-tickers['BitcoinVenezuelaARS'] = null;
-tickers['LocalbitcoinsARS'] = null;
-tickers['CoinbaseARS'] = null;
-tickers['BitexARS'] = null;
-tickers['LocalbitcoinsCLP'] = null;
-tickers['BitsoMXN'] = null;
-tickers['BitXZAR'] = null;
-tickers['CoinbaseZAR'] = null;
-tickers['Bit2CILS'] = null;
-tickers['BTCeLitecoin'] = null;
-tickers['KrakenLitecoin'] = null;
-tickers['VircurexLitecoin'] = null;
-tickers['BTCeLitecoinUSD'] = null;
-tickers['BTCeLitecoinEUR'] = null;
-tickers['VircurexWorldcoin'] = null;
-tickers['CryptsyDogecoin'] = null;
-tickers['KrakenDogecoin'] = null;
-tickers['BTCeNamecoinUSD'] = null;
-tickers['CryptsyAuroracoin'] = null;
-tickers['CryptsyBlackcoin'] = null;
-tickers['CryptsyNxt'] = null;
-tickers['PoloniexNxt'] = null;
-tickers['PoloniexBitshares'] = null;
-tickers['KrakenRipple'] = null;
-tickers['PoloniexMaidsafe'] = null;
-tickers['PoloniexBitcoindark'] = null;
-tickers['PoloniexMonero'] = null;
-tickers['CryptsyDashBTC'] = null;
-tickers['CryptsyDashUSD'] = null;
-tickers['BitFinexDashBTC'] = null;
-tickers['BitFinexDashUSD'] = null;
-tickers['PoloniexBurst'] = null;
-const MAX_TICKERS = Object.keys(tickers).length;
 const DEBUG = false
-var ticker_creators = new Array(); // Store all tickers creators here
-var ordered_tickers = new Array();
-var last_ticker_position = 0;
+const DATA_PROVIDERS_URL = "https://raw.githubusercontent.com/neoranga55/bitcoin-price-ticker/refactor-to-use-firefox-frames/data/data-providers.json"
+const ADDON_UPDATE_DOCUMENT_URL = "http://neoranga55.github.io/bitcoin-price-ticker/"
+
+// The main module of the Add-on.
+var ui = require('sdk/ui')
+const {Cc, Ci, Cu} = require("chrome")
+Cu.import("resource://gre/modules/AddonManager.jsm") // Addon Manager required to know addon version
+const setTimeout = require("sdk/timers").setTimeout
+const setInterval = require("sdk/timers").setInterval
+const clearInterval = require("sdk/timers").clearInterval
+const ADDON_ID = "jid0-ziK34XHkBWB9ezxd4l9Q1yC7RP0@jetpack"
+const DEFAULT_REFRESH_RATE = 60
+const DEFAULT_FONT_SIZE = 14
```

Comment:
'const' is available in ES6 (use esnext option) or Mozilla JS extensions (use moz).<br>Missing semicolon.